hal/armv7m/imxrt/117x/imxrt.c

```diff
@@ -178,11 +178,15 @@ __attribute__((section(".noxip"))) int _imxrt_setIOpad(int pad, char sre, char d
 		t = *reg & ~0x1f;
 		t |= (!!sre) | (!!dse << 1) | (!!pue << 2) | (!!pus << 3);
 
-		if ((pad >= pctl_pad_test_mode) && (pad <= pctl_pad_gpio_snvs_09)) {
+		if ((pad >= pctl_pad_gpio_emc_b1_00) && (pad <= pctl_pad_gpio_disp_b2_15)) {
+			t &= ~(1 << 4);
+			t |= !!ode << 4;
+		}
+		else if ((pad >= pctl_pad_wakeup) && (pad <= pctl_pad_gpio_snvs_09)) {
 			t &= ~(1 << 6);
 			t |= !!ode << 6;
 		}
-		else {
+		else if ((pad >= pctl_pad_gpio_lpsr_00) && (pad <= pctl_pad_gpio_lpsr_15)) {
 			t &= ~(1 << 5);
 			t |= !!ode << 5;
 		}
```

Comment:
`else` error perhaps?

Reply:
Range check is performed in `_imxrt_IOpadGetReg`, perhaps it's ok to replace last `else if` with `else`? Or does some range of pads does not allow `ode` setting?

Reply:
`pctl_pad_test_mode`, `pctl_pad_por_b` and `pctl_pad_onoff` pads' ODE field is read-only

Reply:
I think that error on else after those checks would be a good idea

Reply:
Ok. Sadly, we still need an empty `else` because of MISRA

Reply:
> I think that error on else after those checks would be a good idea

Perhaps I don't understand, but if we can't set `ode` it's ok to just skip it, if we error on `else`, then we will always error on pads that do not have `ode` setting

Reply:
okay, that's valid, I forgot about that. I will try to rewrite it somehow